pyro/primitives.py

```diff
@@ -162,6 +162,13 @@ def factor(name, log_factor):
     Factor statement to add arbitrary log probability factor to a
     probabilisitic model.
 
+    .. warning:: Beware using factor statements in guides. Factor statements
+        assume ``log_factor`` is computed from non-reparametrized statements
+        such as observation statements ``pyro.sample(..., obs=...)``. If
+        instead ``log_factor`` is computed from e.g. the Jacobian determinant
+        of a transformation of a reparametrized variable, factor statements
+        in the guide will result in incorrect results.
```

Comment:
"result in unnecessarily high variance gradients in the context of SVI"?

Reply:
Gradient estimates would be not merely high variance but incorrect.  To make gradients correct users would need to additionally use non-reparametrized versions of distributions, triggering REINFORCE for that site (as we discussed via `fakes.NonreparametrizedNormal`).  Because this contract-at-a-distance is difficult to verify, I think we should recommend assuming gradients are incorrect.